src/frontend/overlay/src/components/molecules/Clock.jsx

```diff
@@ -32,9 +32,16 @@ export const ContestClock = ({
         }
     }, [contestInfo, contestCountdownMode]);
 
+    const getDateTimeNowWithCustomTimeZone = (zone) =>
+        DateTime.now().setZone(zone).toFormat(quietMode ? "HH:mm" : "HH:mm:ss");
+
     const getStatus = useCallback(() => {
-        if (globalTimeMode === true || timeZone !== "") {
-            return DateTime.now().setZone(timeZone !== "" ? timeZone : new SystemZone()).toFormat(quietMode ? "HH:mm" : "HH:mm:ss");
+        if (globalTimeMode === true ) {
+            return getDateTimeNowWithCustomTimeZone(new SystemZone());
+        }
+
+        if (timeZone !== "") {
+            return getDateTimeNowWithCustomTimeZone(timeZone);
```

Comment:
This is the same but with extra steps. Now it's impossible to show global time without setting the timezone. What I meant is clock shouldn't be in global mode when timezone is specified, but globalTimeMode is set.
I think a better fix would be: 
```javascript
if (globalTimeMode === true ) {
            return getDateTimeNowWithCustomTimeZone(timeZone ?? new SystemZone());
}
```
And no default for timezone should be null instead of an empty string.

Reply:
Ok, I think this is a good idea